src/guide/data_types.md

```diff
@@ -60,6 +78,26 @@ subtractOne(counter)
 print(counter) // { count: 1 }
 ```
 
+### Record Spread
+
+Record spread syntax can be used to create a new record value by taking another record value of the same type as a base and overriding some fields. Using the record spread does not mutate the base record.
+
+```grain
+module Main
+
+record Person {
+  name: String,
+  age: Number,
+  favoriteFood: String
+}
+
+let oldMe = { name: "Victor", age: 23, favoriteFood: "Lasagna" }
+// Equivalent to { name: oldMe.name, age: oldMe.age, favoriteFood: "Sushi" }
+let newMe = { ...oldMe, favoriteFood: "Sushi" }
+
+print(newMe) // { name: "Victor", age: 23, favoriteFood: "Sushi" }
+```
+
```

Comment:
I think a really underrated but useful part part of `...` syntax is its a shallow clone and I think explaining that might be nice something like `Record spread syntax can also be used to create a shallow clone of a record.` and then a small example might be nice, maybe in relation to a record with a mutable field.

Reply:
`{ ...a }` is actually not allowed right now, you need to specify at least one field to override